pkg/subctl/cmd/join.go

```diff
@@ -419,6 +445,35 @@ func isOpenShiftCVOEnabled(config *rest.Config) (bool, error) {
 	return scale.Spec.Replicas > 0, nil
 }
 
+func cidrIsPreConfigured(clusterID string, globalNetworks map[string]*globalnet.GlobalNetwork) bool {
+	// GlobalCIDR is not pre-configured
+	if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
+		return false
+	}
+	// GlobalCIDR is pre-configured
+	return true
+}
+
+func validateGlobalnetConfiguration(subctlData *datafile.SubctlData) {
+	if subctlData.GlobalnetCidrRange != "" && globalnetClusterSize != 0 && globalnetClusterSize != subctlData.GlobalnetClusterSize {
+		clusterSize, err := globalnet.GetValidClusterSize(subctlData.GlobalnetCidrRange, globalnetClusterSize)
+		if err != nil || clusterSize == 0 {
+			exitOnError("Invalid globalnet-cluster-size", err)
+		}
+		subctlData.GlobalnetClusterSize = clusterSize
+	}
+
+	if globalnetCIDR != "" && globalnetClusterSize != 0 {
```

Comment:
Similarly, add a check that if subctlData.GlobalnetCidrRange is empty (i.e., Globalnet is not enabled during deploy-broker), but during join operation, user provides globalnet-cluster-size (and/or globalnet-cidr), then these values are ignored.
for example: Globalnet is not enabled, so ignoring globalnet-cluster-size (or globalnet-cidr)

@vthapar @mangelajo  Ignoring should be fine right? Or do you suggest that we exitOnError?

Reply:
I think ignoring with a log line should be fine.

Reply:
Ignore with a log should be fine.